pkg/controllers/provisioningv2/rke2/machineprovision/args.go

```diff
@@ -90,6 +94,11 @@ func (h *handler) getArgsEnvAndStatus(infraObj *infraObject, args map[string]int
 		},
 	}
 
+	machine, err := h.machines.Get(infraObj.meta.GetNamespace(), infraObj.meta.GetName())
```

Comment:
The infrastructure object and the CAPI machine don't have the same name. I think you can do:
```suggestion
	machine, err := rke2.GetMachineByOwner(h.machines, infraObj.meta)
```

Reply:
I have addressed this and I am now using the capi machine object for additional lookups in `getArgsEnvAndStatus` and also passing the machine object into getSecretData to simplify the number of look ups.